core/key.go

```diff
@@ -106,6 +107,19 @@ type CollectionSchemaVersionKey struct {
 
 var _ Key = (*CollectionSchemaVersionKey)(nil)
 
+// SchemaHistoryKey holds the pathway through the schema version history for
+// any given schema.
+//
+// The keys point to the schema version id of the next version of the schema.
+// If a SchemaHistoryKey does not exist for a given SchemaVersionID it means
+// that that SchemaVersionID is for the latest version.
```

Comment:
I find the comment somewhat confusing. It says "schema version id of the next version" but the member's name is `PreviousSchemaVersionId` (btw use "ID" please, not "Id"). Where is this next version? Does `SchemaID` hold this information? Then maybe there is some inconsistency in using terms schema id and schema version id?

Reply:
- [ ] Fix typo `keys point` => `key points`

- [ ] Capitalize `ID`...

> Where is this next version?

The key points to it. The next version is the value component of the store key-value item.

> PreviousSchemaVersionId

Would you prefer just `SchemaVersionID`?

Reply:
~~Not sure. I'm still confused with the comment. It says "next" while the value that is stored is previous~~

oh, I see it's just a key in KV storage that points to the next version. I overlooked this part.
Never mind mind comment then.